packages/lib-utils/src/app/redux/actions/k8s/index.ts

```diff
@@ -0,0 +1,335 @@
+        if (!REF_COUNTS[id]) {
+          consoleLogger.info(`stopped watching ${id} before finishing incremental loading.`);
+          // call cleanup function out of abundance of caution...
+          dispatch(stopK8sWatch(id));
+          return;
+        }
+
+        if (WS[id]) {
+          consoleLogger.warn(`Attempted to create multiple websockets for ${id}.`);
+          return;
+        }
+
+        if (!_.get(k8skind, 'verbs', ['watch']).includes('watch')) {
+          consoleLogger.warn(
+            'Resource does not support watching, falling back to polling.',
+            k8skind,
+          );
+          if (!POLLs[id]) {
+            POLLs[id] = setTimeout(pollAndWatch, 15 * 1000);
+          }
+          return;
+        }
+
+        const { subprotocols } = getImpersonate(getState()) || {};
+        WS[id] = k8sWatch(
+          k8skind,
+          { ...queryWithCluster, resourceVersion },
+          { subprotocols, timeout: 60 * 1000 },
+        );
+      } catch (e) {
+        if (!REF_COUNTS[id]) {
+          consoleLogger.error(
+            `stopped watching ${id} before finishing incremental loading with error ${e}!`,
+          );
+          // call cleanup function out of abundance of caution...
+          dispatch(stopK8sWatch(id));
+          return;
+        }
+
+        dispatch(errored(id, e));
+
+        if (!POLLs[id]) {
+          POLLs[id] = setTimeout(pollAndWatch, 15 * 1000);
+        }
+        return;
+      }
+
+      WS[id]
+        .onclose((event: { code: number }) => {
+          // Close Frame Status Codes: https://tools.ietf.org/html/rfc6455#section-7.4.1
+          if (event.code !== 1006) {
+            return;
+          }
+          consoleLogger.info('WS closed abnormally - starting polling loop over!');
+          const ws = WS[id];
+          ws?.destroy();
+        })
+        .ondestroy((timedOut: boolean) => {
+          if (!timedOut) {
+            return;
+          }
+          // If the WS is unsucessful for timeout duration, assume it is less work
+          //  to update the entire list and then start the WS again
+
+          consoleLogger.info(`${id} timed out - restarting polling`);
+          delete WS[id];
+
+          if (POLLs[id]) {
+            return;
+          }
+
+          POLLs[id] = setTimeout(pollAndWatch, 15 * 1000);
+        })
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        .onbulkmessage((events: any) =>
+          [updateListFromWS, extraAction].forEach((f) => f && dispatch(f(id, events))),
+        );
+    };
+    return pollAndWatch();
+  };
+
+export const watchK8sObject =
+  (
+    id: string,
+    name: string,
+    namespace: string,
+    query: { [key: string]: string },
+    k8sType: K8sModelCommon,
+    partialMetadata = false,
+  ) =>
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  (dispatch: any, getState: any) => {
```

Comment:
```suggestion
  (dispatch: Dispatch, getState: GetState): void => {
```
Maybe we make a `GetState` common type too; we'll definitely have other uses.
```
type GetState = () => SDKStoreState;
```